### PR TITLE
feat: add make extract_translations

### PR DESCRIPTION
**Pull request by shadinaif**
Add `make extract_translations`

This PR needs https://github.com/openedx/RecommenderXBlock/pull/66 before it's merged.

Refs: FC-0012 OEP-58

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @shadinaif!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,8 @@
+WORKING_DIR := recommender
+EXTRACT_DIR := $(WORKING_DIR)/translations/en/LC_MESSAGES
```
Comment (OmarIthawi) [inline Makefile:2]: it should be in `conf/locale` as far as I know. check OEP-58 for more info.

File: recommender/translations/config.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Configuration for i18n workflow.
```
Comment (OmarIthawi) [inline recommender/translations/config.yaml:1]: please remove this file unless absolutely needed

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
@shadinaif this is a good start, but please use the newer OEP-58 structure which needs only generation and no pulling at all.

 - My understanding that config.yaml isn't needed
 - No need for links to `recommender/locale`
 - File structure should match what the github action expects: https://github.com/openedx/openedx-translations/blob/main/.github/workflows/extract-translation-source-files.yml#L85-L102

**Comment by e0d** [comment]
@shadinaif and @OmarIthawi note that we require a conventional commit message format that is described here, https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0051-bp-conventional-commits.html#type

The linter is complaining about the format currently.

File: Makefile (hunk 1 above)
Comment (shadinaif) [reply Makefile]: Fixed. Please note that I had to edit `.tx/config` to do that

File: recommender/translations/config.yaml (hunk 2 above)
Comment (shadinaif) [reply recommender/translations/config.yaml]: it is needed by `i18n_tool`. I've cleaned it to the minimum

File: recommender/conf/locale/en/LC_MESSAGES/text.po [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# SOME DESCRIPTIVE TITLE.
```
Comment (shadinaif) [inline recommender/conf/locale/en/LC_MESSAGES/text.po:1]: @OmarIthawi this is the generated file after running `make extract_translations`

Comment (OmarIthawi) [reply recommender/conf/locale/en/LC_MESSAGES/text.po:1]: Please remove it and add it to `.gitignore`

**Review by OmarIthawi — COMMENTED** [review]
Please use `feat: add make extract_translations` as the commit name and remove the generated file. Otherwise it should be ready.

**Comment by shadinaif** [comment]
all done @OmarIthawi , thank you!

**Review by OmarIthawi — APPROVED** [review]

**Review by brian-smith-tcril — APPROVED** [review]
LGTM :shipit:

**Pull request merged**